### PR TITLE
make linkChild idempotent

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x]  find related tests
- [x]  add a test to ensure we don't regress

while looking at @sandstrom https://github.com/sandstrom/initial-render-components I noticed, copious amounts of u needed changes events where happening. 

That example although synthetic, does indicate death a a thousand papercuts scenario. With a large 15 -> 20% GC cycle. GC is just the result of us allocating to much.

Anyways, back on topic. to many change events where happening.These changes have additional side-affects, but the main one is `sendEvent` itself, which seems to have drop by 15% -> 25% with this change.

I really wouldn't call home about this one, but it helps a teeny tiny bit.

### CODE REVIEW

File: packages/ember-views/lib/mixins/view_child_views_support.js [hunk 1]
```diff
@@ -122,9 +122,11 @@ var ViewChildViewsSupport = Mixin.create({
...
   linkChild(instance) {
-    instance.container = this.container;
-    set(instance, 'parentView', this);
-    instance.trigger('parentViewDidChange');
+    if (get(instance, 'parentView') !== this) {
+      // linkChild should be idempotentj
+      set(instance, 'parentView', this);
+      instance.trigger('parentViewDidChange');
+    }
     instance.ownerView = this.ownerView;
```
Comment (stefanpenner) [inline packages/ember-views/lib/mixins/view_child_views_support.js]: I really think this should only be set at creation.

**Comment by stefanpenner** [comment]
@tomdale / @wycats

**Comment by wycats** [comment]
r+

**Comment by sandstrom** [comment]
Great! I've updated the repo to use this change.

I really appreciate that you are looking into this! It'll make the views-to-components paradigm better in apps with a lot of components.

*(for people stumbling upon this commit, more details can be found here: https://github.com/emberjs/ember.js/issues/11502#issuecomment-119667735)*

**Comment by stefanpenner** [comment]
> I really appreciate that you are looking into this! It'll make the views-to-components paradigm better in apps with a lot of components.

np. The large chunk of this type of waste still needs to be addressed (the attrs object, and related backwards compat) I have been noodling on it, and think there is a way to sort it out.

**Pull request merged**